core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -557,7 +561,8 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
    */
   private def replay(
       eventLog: FileStatus,
-      bus: ReplayListenerBus): Option[FsApplicationAttemptInfo] = {
+      bus: ReplayListenerBus,
+      eventsFilter: (String) => Boolean = (s: String) => true): Option[FsApplicationAttemptInfo] = {
```

Comment:
Does this need to be parameterized? it seems like this method always can apply the filter you pass here. It's called two places.